### PR TITLE
settings.json parent folder functionality

### PR DESCRIPTION
**Pull request by AbdullahAlmanei**
## Summary of the Pull Request

Added menu flyout to "Open JSON" that takes you to the JSON parent folder instead.
This is in response to #12382 
Added comments for the one function I had to implement for the path. Nothing complicated.

### CODE REVIEW

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp
Comment (AbdullahAlmanei) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:1]: Messy formatting, should clean up.

**Comment by AbdullahAlmanei** [comment]
@microsoft-github-policy-service agree

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsEditor/MainPage.cpp:1]: `tools\runformat.cmd` from this repo will help reformat files automatically 😉

File: src/cascadia/TerminalSettingsEditor/MainPage.xaml
Comment (zadjii-msft) [inline src/cascadia/TerminalSettingsEditor/MainPage.xaml:168]: If you give this a `x:Uid`, like:
```xaml
            <muxc:NavigationViewItem x:Name="OpenJsonFolderNavItem"
                                     x:Uid="Nav_OpenJsonFolder"
```

Then you should be able to set the text in the `TerminalSettingsEditor\Resources\en-us\resources.resw` like:

```xml
  <data name="Nav_OpenJsonFolder.Text" xml:space="preserve">
    <value>Open JSON folder</value>
  </data>
```

And even the tooltip should be settable with
```xml
  <data name="Nav_OpenJsonFolder.[using:Windows.UI.Xaml.Controls]ToolTipService.ToolTip" xml:space="preserve">
    <value>Open the folder holding your settings.json file.</value>
  </data>
```

And if that doesn't work, then there's something wacky going on with XAML

**Review by zadjii-msft — COMMENTED** [review]
Overall, I'm cool with this. If you get really stuck with the Resources thing, then we can always just merge as-is and have the core team fix in post ☺️ 

(sorry for the delays, we were all a little preoccupied with Build last week)

File: src/cascadia/TerminalSettingsEditor/MainPage.xaml
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/MainPage.xaml:168]: Hmm... Mind committing your changes anyways? I tried the suggested changes on my end, and it worked fine. I'd be happy to take a closer look at your approach and let you know how to fix it :)

Also, you're not referencing the name `OpenJsonFolderNavItem` anywhere, so you might as well remove it.

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Good progress! Just a few things before I sign off.

I'm also going to add the "Needs-Discussion" tag so we discuss this as a team. Specifically...
- I have concerns that this feature might not be discoverable
- Should we update the "new tab dropdown" "open settings" item so that if you click it with a specific modifier, it'll open the folder.
   - Should the same be done for the settings button in the settings UI (the one you're messing with right now), instead of adding a flyout?

Feel free to brainstorm and discuss in this PR too. :)

File: src/cascadia/TerminalSettingsEditor/MainPage.xaml
Comment (AbdullahAlmanei) [reply src/cascadia/TerminalSettingsEditor/MainPage.xaml:168]: It's a relic of me trying to get the resources to work! Curious that it works on your end. Must be my system. I'll commit the changes, maybe I am doing something wrong, too. I appreciate the feedback!

**Comment by AbdullahAlmanei** [comment]
> - I have concerns that this feature might not be discoverable

I definitely agree; you'd have to right click by accident to discover it the way it's set up right now. 

> - Should we update the "new tab dropdown" "open settings" item so that if you click it with a
specific modifier, it'll open the folder.

The thing is, there's already and alt-click modifier currently. Wouldn't the tooltip get quite lengthy if we tried something like that?

>    - Should the same be done for the settings button in the settings UI (the one you're messing with right now), instead of adding a flyout?

I do believe that we should apply whatever we decide is best universally. As in, it should be the same anywhere you can access the settings JSON.


Really, I think our options are either a modifier or some kind of drop down. The latter might result  in a messy user experience . A possible approach to the former might to figure out some other place we can move the current modifier to. This is, of course, if we believe that the JSON folder is a more significant and frequent operation than accessing the default JSON file. (I believe that's the current applied modifier, I have to double-check.)

**Comment by AbdullahAlmanei** [comment]
@carlos-zamora I committed the changes I talked about. However, do note that it's still not working on my local system.  I have the output debug text file where the program crashes when I try to navigate to the settings menu at all. I have attached it here.
[Output-Debug.txt](https://github.com/microsoft/terminal/files/11713614/Output-Debug.txt)

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:581]:
```suggestion
	<data name="Nav_OpenJSONFolder.Text" xml:space="preserve">
```
Found the bug! This'll fix the crash!

File: src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw
Comment (carlos-zamora) [inline src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:580]: Also, make sure you use spaces here, not tabs.

If you want, you can also edit this file in Visual Studio. It gives you a table to work with, and always formats it correctly.

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Regarding my comment earlier about discoverability. We chatted about this PR more as a team. Two things came out of it:
1. We feel it'd be best to prioritize this as an `action` and add it to the command palette. To do that, you'll have to...
   - follow the instructions from [this comment](https://github.com/microsoft/terminal/pull/15417/files?show-viewed-files=true&file-filters%5B%5D=#r1223382777). This hooks it up as an action.
   - add `{ "command": { "action": "openSettings", "target": "<JSON KEY FROM TerminalSettingsSerializationHelpers.h>" } },` to defaults.json. This adds it to the command palette for everyone. I'd add it next to the other `openSettings` actions.
2. Since the context menu isn't very discoverable, we think long-term, a better design would be something like the following:
   - alternate design with folder icon appearing on hover
   - That way the folder icon appears on hover.
   - However, this is a much more tedious to do and will have accessibility concerns/issues. It's doable, but it should definitely be its own standalone PR.

That said, we'd like to pivot this PR to focus on accomplishing the first item above and remove the context menu functionality (so basically, revert all changes made to `src/cascadia/TerminalSettingsEditor`). Then, as a follow-up, the UI changes can be introduced with the design from the second point above.

How does that sound? Happy to help and provide any more guidance along the way. 😊

**Pull request closed without merging**
